syntax = "proto3";

package resemble.v1alpha1;

////////////////////////////////////////////////////////////////////////
// Resemble:
////////////////////////////////////////////////////////////////////////

message ActorAlreadyConstructed {}

////////////////////////////////////////////////////////////////////////

message ActorNotConstructed {
  bool requires_constructor = 1;
}

////////////////////////////////////////////////////////////////////////

message UnknownService {}

////////////////////////////////////////////////////////////////////////

message UnknownTask {}

////////////////////////////////////////////////////////////////////////

message TransactionParticipantFailedToPrepare {}

////////////////////////////////////////////////////////////////////////

message TransactionParticipantFailedToCommit {}

////////////////////////////////////////////////////////////////////////
// gRPC:
////////////////////////////////////////////////////////////////////////

message Cancelled {}

////////////////////////////////////////////////////////////////////////

message Unknown {}

////////////////////////////////////////////////////////////////////////

message InvalidArgument {}

////////////////////////////////////////////////////////////////////////

message DeadlineExceeded {}

////////////////////////////////////////////////////////////////////////

message NotFound {}

////////////////////////////////////////////////////////////////////////

message AlreadyExists {}

////////////////////////////////////////////////////////////////////////

message PermissionDenied {}

////////////////////////////////////////////////////////////////////////

message ResourceExhausted {}

////////////////////////////////////////////////////////////////////////

message FailedPrecondition {}

////////////////////////////////////////////////////////////////////////

message Aborted {}

////////////////////////////////////////////////////////////////////////

message OutOfRange {}

////////////////////////////////////////////////////////////////////////

message Unimplemented {}

////////////////////////////////////////////////////////////////////////

message Internal {}

////////////////////////////////////////////////////////////////////////

message Unavailable {}

////////////////////////////////////////////////////////////////////////

message DataLoss {}

////////////////////////////////////////////////////////////////////////

message Unauthenticated {}

////////////////////////////////////////////////////////////////////////
